cloud/ec2.go

```diff
@@ -687,6 +689,37 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 				"host_provider": h.Distro.Provider,
 				"distro":        h.Distro.Id,
 			}))
+			if h.ShouldFallbackToOnDemand() && (strings.Contains(err.Error(), EC2InsufficientCapacity) || strings.Contains(err.Error(), EC2UnfulfillableCapacity)) {
```

Comment:
I would just double check that there isn't a case where we want to spawn a host but ignore `ShouldFallbackToOnDemand()` regardless if it returns true. @ybrill would know better than I do.

Reply:
I don't think so.